### PR TITLE
[Access] Implement conversion functions for CCF

### PR DESCRIPTION
**Pull request by koko1123**
Access node side conversion functions for CCF encoding

Todo:
- merge: https://github.com/onflow/flow/pull/1343
- use conversion functions in access API (replace uses of `MessagesToEvents` with `MessagesToEventsFromFormat`

### CODE REVIEW

File: engine/common/rpc/convert/convert.go [hunk 1]
```diff
@@ -645,14 +658,66 @@ func MessagesToEvents(l []*entities.Event) []flow.Event {
...
+				m.EventIndex, version, err)
+		}
+		events[i] = *event
+	}
+	return events, nil
+}
+
+func MessageToEventFromVersion(m *entities.Event, version execproto.EventEncodingVersion) (*flow.Event, error) {
+	// CCF
+	if version == execproto.EventEncodingVersion_CCF_V0 {
```
Comment (janezpodhostnik) [inline engine/common/rpc/convert/convert.go:676]: might be better to use switch case?

File: engine/access/rpc/backend/backend_transactions.go [hunk 2]
```diff
@@ -410,10 +410,15 @@ func (b *backendTransactions) GetTransactionResultsByBlockID(
 				return nil, rpc.ConvertStorageError(err)
 			}
 
+			events, err := convert.MessagesToEventsFromVersion(txResult.GetEvents(), txResult.GetEventEncodingVersion())
```
Comment (janezpodhostnik) [inline engine/access/rpc/backend/backend_transactions.go:413]: 4x repeated block with the same error message string. Consider refactoring.

**Review by janezpodhostnik — APPROVED** [review]
Left some suggestions

**Review by zhangchiqing — APPROVED** [review]

**Review by fxamacker — APPROVED** [review]
Thanks for implementing this! :+1:   I mainly focused on conversion.

**Pull request merged**